### PR TITLE
Task 2 and 3

### PR DESCRIPTION
**Pull request by trave11**
Changes
1. Refactoring,
2. New validation during product creation and search by id.

### CODE REVIEW

File: src/com/javaguru/shoppinglist/repository/ProductRepository.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    public Long getProductIdSequence() {
+        return productIdSequence;
+    }
+
+    public boolean existsByName(String providedName) {
+        boolean isFound = false;
+        for (Product product : productRepository.values()) {
+            if (product.getName().equals(providedName)) {
+                isFound = true;
```
Comment (javagururt) [inline src/com/javaguru/shoppinglist/repository/ProductRepository.java]: можно сразу сделать return true

File: src/com/javaguru/shoppinglist/service/validation/rules/ProductNameValidationRule.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    private final ProductRepository productRepository;
+
+    public ProductNameValidationRule(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @Override
+    public void validate(Product product) {
+        checkNotNull(product);
+        if (product.getName().length() < 3) {
```
Comment (javagururt) [inline src/com/javaguru/shoppinglist/service/validation/rules/ProductNameValidationRule.java]: 3 & 32 magic numbers

File: src/com/javaguru/shoppinglist/repository/ProductRepository.java (hunk 1 above)
Comment (trave11) [reply src/com/javaguru/shoppinglist/repository/ProductRepository.java]: fixed

File: src/com/javaguru/shoppinglist/service/validation/rules/ProductNameValidationRule.java (hunk 2 above)
Comment (trave11) [reply src/com/javaguru/shoppinglist/service/validation/rules/ProductNameValidationRule.java]: fixed

**Pull request merged**